winpr/libwinpr/environment/environment.c

```diff
@@ -645,3 +645,71 @@ char** EnvironmentBlockToEnvpA(LPCH lpszEnvironmentBlock)
 
 	return envp;
 }
+
+#ifdef _WIN32
+
+DWORD GetEnvironmentVariableX(char* lpName, char* lpBuffer, DWORD nSize)
+{
+	int status;
+	DWORD result = 0;
+	DWORD nSizeW = 0;
+	LPWSTR lpNameW = NULL;
+	LPWSTR lpBufferW = NULL;
+	LPSTR lpBufferA = lpBuffer;
+
+	if (ConvertToUnicode(CP_UTF8, 0, lpName, -1, &lpNameW, 0) < 1)
+		goto cleanup;
+
+	if (!lpBuffer)
+	{
+		char lpBufferMaxA[2048];
+		WCHAR lpBufferMaxW[2048];
+
+		lpBufferA = lpBufferMaxA;
+		lpBufferW = lpBufferMaxW;
+		nSizeW = sizeof(lpBufferMaxW) / 2;
+
+		result = GetEnvironmentVariableW(lpNameW, lpBufferW, nSizeW);
+
+		status = ConvertFromUnicode(CP_UTF8, 0, lpBufferW, -1, &lpBufferA, sizeof(lpBufferMaxA),
+		                            NULL, NULL);
+
+		if (status > 0)
+			result = (DWORD)status;
+
+		return result;
+	}
+	else
+	{
+		nSizeW = (nSize + 1) * 2;
+		lpBufferW = calloc(1, nSizeW);
```

Comment:
In the first block, nSizeW is a wide character count. In the second block, it is a byte count.

Maybe make this more consistent like this:
```
nSizeW = nSize + 1;
lpBufferW = calloc(nSizeW, 2);
...
result = GetEnvironmentVariableW(lpNameW, lpBufferW, nSizeW);
```

Reply:
I fixed it, thanks!